File: VB/CustomEditSettingExample/ViewModel/TaskViewModel.vb
Imports System
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace CustomEditSettingExample
	Friend Class TaskViewModel
		Private _List As ObservableCollection(Of Task)
		Public ReadOnly Property List() As ObservableCollection(Of Task)
			Get
				If _List Is Nothing Then
					_List = New ObservableCollection(Of Task)()
					For i As Integer = 0 To 99
						_List.Add(New Task() With {
							.Name = "Task" & i,
							.Date = Date.Now.AddDays((New Random(i)).Next(1, 31))
						})
					Next i
				End If
				Return _List
			End Get
		End Property

		Public Class Task
			Public Property Name() As String
			Public Property [Date]() As Date
		End Class
	End Class
End Namespace

File: CS/CustomEditSettingExample/ViewModel/TaskViewModel.cs
using System;
using System.Collections.ObjectModel;
using System.Linq;

namespace CustomEditSettingExample
{
    class TaskViewModel
    {
        private ObservableCollection<Task> _List;
        public ObservableCollection<Task> List {
            get {
                if (_List == null) {
                    _List = new ObservableCollection<Task>();
                    for (int i = 0; i < 100; i++) {
                        _List.Add(new Task() { Name = "Task" + i, Date = DateTime.Now.AddDays(new Random(i).Next(1, 31)) });
                    }
                }
                return _List;
            }
        }

        public class Task
        {
            public string Name { get; set; }
            public DateTime Date { get; set; }
        }
    }
}
